pkg/templatelookup/regular_test.go

```diff
@@ -1139,13 +1091,53 @@ func TestTemplateNameMatch(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := templatelookup.TemplateNameMatch(&tt.template, targetName); got != tt.want {
+			if got := moduletemplateinfolookup.TemplateNameMatch(&tt.template, targetName); got != tt.want {
 				assert.Equal(t, tt.want, got)
 			}
 		})
 	}
 }
 
+type getRegularTemplatesTestCases []struct {
+	name            string
+	kyma            *v1beta2.Kyma
+	wantVersion     string
+	wantChannel     string
+	wantErrContains string
+}
+
+func executeGetRegularTemplatesTestCases(t *testing.T,
+	testCases getRegularTemplatesTestCases,
+	availableModuleTemplates v1beta2.ModuleTemplateList,
+	availableModuleReleaseMetas v1beta2.ModuleReleaseMetaList,
+	moduleToInstall v1beta2.Module,
+) {
+	t.Helper()
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			reader := NewFakeModuleTemplateReader(availableModuleTemplates, availableModuleReleaseMetas)
```

Comment:
We don't need constructor functions for mock impls. Also can be renamed to `MockReader` or `MockModuleTemplateReader`